lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -5066,7 +5064,14 @@ struct ExportVerilogPass : public ExportVerilogBase<ExportVerilogPass> {
     // TODO: This should be moved up to circt-opt and circt-translate.
     applyLoweringCLOptions(getOperation());
 
-    if (failed(exportVerilog(getOperation(), os)))
+    // Prepare the ops in the module for emission.
+    mlir::OpPassManager preparePM("builtin.module");
+    auto &modulePM = preparePM.nest<hw::HWModuleOp>();
+    modulePM.addPass(createPrepareForEmissionPass());
+    if (failed(runPipeline(preparePM, getOperation())))
+      signalPassFailure();
```

Comment:
```suggestion
      return signalPassFailure();
```
(don't continue execution if there's an error)